src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynSonarLint.cs

```diff
@@ -28,13 +28,13 @@ namespace SonarScanner.MSBuild.PreProcessor.Roslyn
 {
     internal static class RoslynSonarLint
     {
-        public static string GenerateXml(IEnumerable<SonarRule> activeRules, IDictionary<string, string> serverSettings,
+        public static string GenerateXml(IEnumerable<SonarRule> activeRules, IAnalysisPropertyProvider analysisProperties,
```

Comment:
This class generates the `SonarLint.xml` file. Previously, it was just using a list of settings fetched directly from the server.
It should have been aggregating the server settings with values from the command line etc.
This change just makes the class use the property provider interface rather than a simple list.